### PR TITLE
DT-1642: ✨ Add validation to create group

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt [hunk 1]
```diff
@@ -148,17 +151,19 @@ class GroupsService(
   }
 
   @Transactional(transactionManager = "authTransactionManager")
-  @Throws(GroupExistsException::class, GroupNotFoundException::class)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt:151]: fix from previous PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt [hunk 2]
```diff
@@ -111,14 +111,17 @@ class GroupsService(
   @Transactional(transactionManager = "authTransactionManager")
   @Throws(ChildGroupExistsException::class, GroupNotFoundException::class)
   fun createChildGroup(username: String, createChildGroup: CreateChildGroup) {
-    val childGroupFromDB = childGroupRepository.findByGroupCode(createChildGroup.groupCode)
+    val groupCode = createChildGroup.groupCode.trim().toUpperCase()
+    val childGroupFromDB = childGroupRepository.findByGroupCode(groupCode)
     if (childGroupFromDB != null) {
```
Comment (JamieDraperUK) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt:116]: Missed opportunity to use let :grin: .

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt (hunk 2 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/GroupsService.kt:116]: agreed :joy:

**Pull request merged**